File: js/src/vista-sdk/types/GmodDto.ts
export interface GmodDto {
    visRelease: string;
    items: GmodNodeDto[];
    relations: GmodRelationDto[];
}

export interface GmodRelationDto {
    parent: string;
    child: string;
}

export interface GmodNodeDto {
    id?: string;
    category: string;
    type: string;
    code: string;
    name: string;
    commonName?: string | null;
    definition?: string | null;
    commonDefinition?: string | null;
    installSubstructure?: boolean | null;
    normalAssignmentNames?: { [key: string]: string };
    location?: string;
}

File: python/vista_sdk/GmodDto.py
from typing import Dict, List, Optional

from pydantic import BaseModel, Field


class GmodNodeDto(BaseModel):
    category: str = Field(..., alias="category")
    type: str = Field(..., alias="type")
    code: str = Field(..., alias="code")
    name: str = Field("", alias="name")
    common_name: Optional[str] = Field(None, alias="commonName")
    definition: Optional[str] = Field(None, alias="definition")
    common_definition: Optional[str] = Field(None, alias="commonDefinition")
    install_substructure: Optional[bool] = Field(None, alias="installSubstructure")
    normal_assignment_names: Optional[Dict[str, str]] = Field(
        None, alias="normalAssignmentNames"
    )


class GmodDto(BaseModel):
    vis_version: str = Field(..., alias="visRelease")
    items: List[GmodNodeDto] = Field(..., alias="items")
    relations: List[List[str]] = Field(..., alias="relations")
